packages/nimble-components/src/table/index.ts

```diff
@@ -267,9 +268,11 @@ export class Table<
         await this.processPendingUpdates();
 
         const data = newData.map(record => {
-            return { ...record };
+            return { data: { ...record } } as InternalTableRecord<TData>;
```

Comment:
shouldn't need the as cast, the fn provided to the map should just specify a return type